features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/des/des_alt.c

```diff
@@ -14,33 +14,32 @@
  * limitations under the License.
  */
 
-#if !defined(MBEDTLS_CONFIG_FILE)
-#include "mbedtls/config.h"
-#else
-#include MBEDTLS_CONFIG_FILE
-#endif
+#include "mbedtls/des.h"
 
 #if defined(MBEDTLS_DES_C)
 #if defined(MBEDTLS_DES_ALT)
 
 #include <string.h>
-#include "mbedtls/des.h"
-#include "des_alt.h"
+#include <stdbool.h>
 #include "crypto-misc.h"
 #include "nu_bitutil.h"
 #include "mbed_toolchain.h"
+#include "mbed_error.h"
 
-// Must be a multiple of 64-bit block size
+/* DMA buffer
+ * 
+ * MAXSIZE_DMABUF must be a multiple of 64-bit block size.
+ * Its value is estimated to trade memory footprint off against performance. 
+ */
 #define MAXSIZE_DMABUF  (8 * 5)
-static uint8_t dmabuf_in[MAXSIZE_DMABUF] MBED_ALIGN(4);
-static uint8_t dmabuf_out[MAXSIZE_DMABUF] MBED_ALIGN(4);
+MBED_ALIGN(4) static uint8_t dmabuf_in[MAXSIZE_DMABUF];
+MBED_ALIGN(4) static uint8_t dmabuf_out[MAXSIZE_DMABUF];
```

Comment:
I was thinking that if both DES and AES are enabled, then we will be allocating the DMA buffer twice. In practice, this means that you can issue DES and AES simultaneously. However,
* Would the two operations really happen in parallel?
* Is it common at all for an MCU to issue both DES and AES operations in parallel?
I am wondering whether this is an unnecessary cost to pay in terms of space for very little benefit. Also, please that this is just a thought, not a change request.

Reply:
@andresag01 It depends on upper layer. But I think AES/DES H/W acceleration port cannot make such assumption because upper layer can trigger AES/DES in parallel for e.g. performance.

Reply:
@ccli8 Could you point out where it is ensured that these globals are indeed placed in the DMA region `0x20000000 - 0x30000000`?